File: library-infrastructure/src/main/java/com/dodecaedro/library/infrastructure/search/BookToSearchDocumentConverter.java
package com.dodecaedro.library.infrastructure.search;

import com.dodecaedro.library.domain.model.Book;
import org.springframework.stereotype.Component;

@Component
public class BookToSearchDocumentConverter {
	public BookSearchDocument toSearchDocument(Book book) {
		return BookSearchDocument.builder()
			.id(book.getId())
			.title(book.getTitle())
			.build();
	}
}

File: library-infrastructure/src/main/kotlin/com/dodecaedro/library/infrastructure/search/BookToSearchDocumentConverter.kt
package com.dodecaedro.library.infrastructure.search

import com.dodecaedro.library.domain.model.Book
import org.springframework.stereotype.Component

@Component
open class BookToSearchDocumentConverter {
	fun toSearchDocument(book: Book): BookSearchDocument? = BookSearchDocument(id = book.id, title = book.title)
}
